src/chains/celo/fees.test.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, expect, test, vi } from 'vitest'
+import { http, createTestClient } from '~viem/index.js'
+import { celo } from '../index.js'
+
+const client = createTestClient({
+  transport: http(),
+  chain: celo,
+  mode: 'anvil',
+})
+
+describe('celo/fees', () => {
+  test("doesn't call the client when feeCurrency is not provided", async () => {
+    const requestMock = vi.spyOn(client, 'request')
+
+    expect(celo.fees.estimateFeesPerGas).toBeTypeOf('function')
+
+    // @ts-ignore
+    const fees = await celo.fees.estimateFeesPerGas({
+      client,
+      request: {},
+    } as any)
+
+    expect(fees).toBeNull()
+    expect(requestMock).not.toHaveBeenCalled()
+  })
+
+  test('calls the client when feeCurrency is provided', async () => {
+    const requestMock = vi.spyOn(client, 'request')
+    requestMock.mockImplementation((request) => {
+      switch (request.method) {
+        case 'eth_gasPrice':
+          return '11619349802'
+        case 'eth_maxPriorityFeePerGas':
+          return '2323869960'
+      }
+    })
+
+    expect(celo.fees.estimateFeesPerGas).toBeTypeOf('function')
+
+    // @ts-ignore
+    const fees = await celo.fees.estimateFeesPerGas({
+      client,
+      request: {
+        feeCurrency: '0xfee',
+      },
+    } as any)
```

Comment:
hmm we gotta be able to use this without the any and ts-ignore

Reply:
I removed `@ts-ignore` 👍 

as for the `any` it provides simplicity in the tests and you can pass only properties you are interested with and not having to worry about all the stuff you don't need, it's a common practise in the repo as well: https://github.com/wagmi-dev/viem/blob/9b9c308888ef5977c55160426827ad75de686201/src/actions/public/estimateFeesPerGas.test.ts#L266

Reply:
ah ok if its done in other parts all good